contracts/FlyoverDiscovery.sol

```diff
@@ -0,0 +1,266 @@
+    }
+
+    /// @notice Updates the caller LP metadata
+    /// @dev Reverts if the caller is not registered or provides invalid fields
+    /// @param name New LP name
+    /// @param apiBaseUrl New LP API base URL
+    function updateProvider(string calldata name, string calldata apiBaseUrl) external {
+        if (bytes(name).length < 1 || bytes(apiBaseUrl).length < 1) revert InvalidProviderData(name, apiBaseUrl);
+        Flyover.LiquidityProvider storage lp;
+        address providerAddress = msg.sender;
+        for (uint i = 1; i <= lastProviderId; ++i) {
+            lp = _liquidityProviders[i];
+            if (providerAddress == lp.providerAddress) {
+                lp.name = name;
+                lp.apiBaseUrl = apiBaseUrl;
+                emit IFlyoverDiscovery.ProviderUpdate(providerAddress, lp.name, lp.apiBaseUrl);
+                return;
+            }
+        }
+        revert Flyover.ProviderNotRegistered(providerAddress);
+    }
+
+        /// @notice Sets the minimum collateral threshold per service side
+        /// @dev Only callable by DEFAULT_ADMIN_ROLE
+        /// @param minCollateral New minimum collateral value
+        function setMinCollateral(uint minCollateral) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        _minCollateral = minCollateral;
+    }
+
+    /// @notice Lists LPs that should be visible to users
+    /// @dev A provider is listed if it has sufficient collateral for at least one side and `status` is true
+    /// @return providersToReturn Array of LP records to display
+    function getProviders() external view returns (Flyover.LiquidityProvider[] memory) {
+        uint count = 0;
+        Flyover.LiquidityProvider storage lp;
+        for (uint i = 1; i <= lastProviderId; ++i) {
+            if (_shouldBeListed(_liquidityProviders[i])) {
+                ++count;
+            }
+        }
+        Flyover.LiquidityProvider[] memory providersToReturn = new Flyover.LiquidityProvider[](count);
+        count = 0;
+        for (uint i = 1; i <= lastProviderId; ++i) {
+            lp = _liquidityProviders[i];
+            if (_shouldBeListed(lp)) {
+                providersToReturn[count] = lp;
+                ++count;
+            }
+        }
+        return providersToReturn;
+    }
+
+    /// @notice Returns a single LP by address
+    /// @param providerAddress The LP address
+    /// @return provider LP record, reverts if not found
+    function getProvider(address providerAddress) external view returns (Flyover.LiquidityProvider memory) {
+        return _getProvider(providerAddress);
+    }
+
+    /// @notice Checks if an LP can operate for peg-in side
+    /// @dev Ignores the first argument as compatibility stub with legacy signature
+    /// @param addr The LP address
+    /// @return isOp True if registered and peg-in collateral >= min
+    function isOperational(Flyover.ProviderType, address addr) external view returns (bool) {
```

Comment:
I think we should use the `isOperational` version used in the split PoC (the one that absorbs `isOperational` and `isOperationalForPegout` in the same function), as it is easier to consume from other contracts / off chain components

https://github.com/rsksmart/liquidity-bridge-contract/blob/lbc-split/contracts/split/FlyoverDiscovery.sol#L111